opentelemetry-extra/src/OpenTelemetry/EventlogStreaming_Internal.hs

```diff
@@ -298,20 +304,23 @@ handleOpenTelemetryEventlogEvent m st (tid, now, m_trace_id) =
                     spanNanosecondsSpentInGC = 0,
                     spanParentId = parent
                   }
-           in (createSpan span_id sp st', [], [])
+           in (createSpan span_id sp st'', [], [])
         Just span_id ->
           let (st', sp) = emitSpan serial span_id st
-           in (st', [sp {spanFinishedAt = now}], [])
+              (st'', display_tid) = inventDisplayTid tid st'
```

Comment:
If I comment out line 310 my trace no longer crashes tracy. 

Why invent a new displayTid instead of reusing the same one used in the BeginSpan frame? If the events are slightly out of order and the `StopThread` event is processed before the `EndSpan` event, the `displayTid` used here will be completely wrong and crash Tracy.

Reply:
Good catch.